static/js/public/snap-details/channelMap.js

```diff
@@ -1,294 +1,418 @@
+      if (this.openDesktopAttempt > 1) {
+        label = 'Snap install click';
+        value += ` - click ${this.openDesktopAttempt}`;
+      }
+
+      ga('gtm1.send', {
+        hitType: 'event',
+        eventCategory: 'Snap details',
+        eventAction: 'Click view in desktop store button',
+        eventLabel: label,
+        eventValue: value
+      });
+    }
+
+    this.openDesktopAttempt += 1;
   }
-}
 
-function initTabs(el) {
-  el.addEventListener('click', (event) => {
-    const target = event.target.closest('.p-tabs__link');
+  selectScreen(screenEl) {
+    const selected = screenEl.getAttribute('aria-selected');
+    const currentlySelected = screenEl.parentNode.querySelector('.is-open');
 
-    if (target) {
-      event.preventDefault();
-      selectTab(target, el);
+    if (currentlySelected) {
+      currentlySelected.classList.remove('is-open');
     }
-  });
-}
 
-function initOpenSnapButtons() {
-  let attempt = 1;
+    if (!selected) {
+      screenEl.classList.add('is-open');
+      screenEl.setAttribute('aria-selected', "true");
+    }
+  }
 
-  document.addEventListener('click', (event) => {
-    const openButton = event.target.closest('.js-open-snap-button');
+  slideToVersions(clickEl) {
+    const slides = clickEl.closest('.p-channel-map__slides');
+    slides.classList.add('show-left');
+    slides.classList.remove('show-right');
+  }
 
-    if (openButton) {
-      const name = openButton.dataset.snap;
-      let iframe = document.querySelector('.js-snap-open-frame');
+  slideToInstructions(clickEl) {
+    // Add content to the right slide area
+    this.writeInstallInstructions(clickEl.dataset.channel);
 
-      if (iframe) {
-        iframe.parentNode.removeChild(iframe);
-      }
+    const slides = clickEl.closest('.p-channel-map__slides');
+    slides.classList.add('show-right');
+    slides.classList.remove('show-left');
+  }
 
-      iframe = document.createElement('iframe');
-      iframe.className = 'js-snap-open-frame';
-      iframe.style.position = 'absolute';
-      iframe.style.top = '-9999px';
-      iframe.style.left = '-9999px';
-      iframe.src = `snap://${name}`;
-      document.body.appendChild(iframe);
-
-      if (typeof ga !== 'undefined') {
-        // The first attempt should be counted towards the 'intent'
-        let label = 'Snap install intent';
-        let value = `${name}`;
-
-        // Subsequent attempts should still be tracked, but not as 'intent'
-        if (attempt > 1) {
-          label = 'Snap install click';
-          value += ` - click ${attempt}`;
-        }
+  writeInstallInstructions(channel) {
+    let paramString = '';
 
-        ga('gtm1.send', {
-          hitType: 'event',
-          eventCategory: 'Snap details',
-          eventAction: 'Click view in desktop store button',
-          eventLabel: label,
-          eventValue: value
-        });
+    if (channel.indexOf('latest') === 0) {
```

Comment:
I have no idea what this is trying to do... Index equals 0? or 7? ... is channel a table or string? why are we looking for such specific index of substrings?